Upstream: https://github.com/neomutt/neomutt/pull/452
--
From ff8eaec57f5a4cb9f67e4391fa2024be9569bd2b Mon Sep 17 00:00:00 2001
From: Kevin Decherf <kevin@kdecherf.com>
Date: Sat, 4 Mar 2017 20:05:12 +0100
Subject: [PATCH] index_format: add support of %K

Unlike %B, %K will remain empty if there is no list detected for the
message.

Signed-off-by: Kevin Decherf <kevin@kdecherf.com>
---
 hdrline.c | 11 ++++++++++-
 init.h    |  1 +
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/hdrline.c b/hdrline.c
index f72cfc3b..47eb6e1d 100644
--- a/hdrline.c
+++ b/hdrline.c
@@ -422,7 +422,7 @@ static char *apply_subject_mods (ENVELOPE *env)
 /* %a = address of author
  * %A = reply-to address (if present; otherwise: address of author
  * %b = filename of the originating folder
- * %B = the list to which the letter was sent
+ * %B = the list to which the letter was sent, or else the folder name (%b).
  * %c = size of message in bytes
  * %C = current message number
  * %d = date and time of message using $date_format and sender's timezone
@@ -435,6 +435,7 @@ static char *apply_subject_mods (ENVELOPE *env)
  * %g = message labels (e.g. notmuch tags)
  * %i = message-id
  * %I = initials of author
+ * %K = the list to which the letter was sent (if any; otherwise: empty)
  * %l = number of lines in the message
  * %L = like %F, except `lists' are displayed first
  * %m = number of messages in the mailbox
@@ -526,6 +527,7 @@ hdr_format_str (char *dest,
       break;
 
     case 'B':
+    case 'K':
       if (!first_mailing_list (dest, destlen, hdr->env->to) &&
 	  !first_mailing_list (dest, destlen, hdr->env->cc))
 	dest[0] = 0;
@@ -535,6 +537,13 @@ hdr_format_str (char *dest,
 	mutt_format_s (dest, destlen, prefix, buf2);
 	break;
       }
+      if (op == 'K')
+      {
+        if (optional)
+          optional = 0;
+        /* break if 'K' returns nothing */
+        break;
+      }
       /* fall through if 'B' returns nothing */
 
     case 'b':
diff --git a/init.h b/init.h
index fcc320f8..5f9da0b6 100644
--- a/init.h
+++ b/init.h
@@ -1487,6 +1487,7 @@ struct option_t MuttVars[] = {
   ** .dt %g .dd message labels (e.g. notmuch tags)
   ** .dt %H .dd spam attribute(s) of this message
   ** .dt %i .dd message-id of the current message
+  ** .dt %K .dd the list to which the letter was sent (if any; otherwise: empty).
   ** .dt %l .dd number of lines in the message (does not work with maildir,
   **            mh, and possibly IMAP folders)
   ** .dt %L .dd If an address in the ``To:'' or ``Cc:'' header field matches an address
-- 
2.11.1

